independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeResourceDeployment.java

```diff
@@ -432,6 +437,45 @@ public RuntimeResource buildResourceMethod(ResourceClass clazz,
                 pathParameterIndexes, score, sseElementType, clazz.resourceExceptionMapper());
     }
 
+    private void addResponseHandler(ServerResourceMethod method, List<ServerRestHandler> handlers) {
```

Comment:
Why do we need the new `ResponseHandler.ResponseBuilderCustomizer` and `PublisherResponseHandler.StreamingResponseCustomizer` integration classes? 

Couldn't we just register a handler in `HandlerChainCustomizer.Phase#AFTER_RESPONSE_CREATED` and use it to update the response?

Reply:
2 reasons:

1) we don't want to update the response if it is created by the user - a Response returned by the user always takes precedence over the annotations.
2) In the case of streaming, there is no such phase. The streaming handler is essentially the last handler to be executed.

Reply:
For 1 we could add an extra check and forgo the use of the customizer, but it would also mean that the Response would be created again - so slightly more overhead.